### PR TITLE
Fix Error class deserialization

### PR DESCRIPTION
**Pull request by ankitdimania-eligible**

### CODE REVIEW

File: src/main/java/com/eligible/json/deserializer/ErrorDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            throws JsonParseException {
+
+        final Gson gson = APIResource.GSON;
+
+        // Dates could be a list of Date.class, or a list of list of Date.class. Merging all list into one
+        if (json.isJsonObject()) {
+            JsonObject errorJson = json.getAsJsonObject();
+            com.eligible.model.Error error = gson.fromJson(json, com.eligible.model.Error.class);
+
+            JsonElement fuac = errorJson.get("follow-up_action_code");
```
Comment (ankit819) [inline src/main/java/com/eligible/json/deserializer/ErrorDeserializer.java]: @ankitdimania-eligible should it be follow_up_action_code?

Comment (ankitdimania-eligible) [inline src/main/java/com/eligible/json/deserializer/ErrorDeserializer.java]: follow_up_action_code is already taken care by gson. This is to cover both the scenarios. That's why this code.

File: src/test/resources/com/eligible/model/api_error_response.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "known_issues": [],
+  "error": {
+    "response_code": "Y",
+    "response_description": "Yes",
+    "agency_qualifier_code": "",
+    "agency_qualifier_description": "",
+    "reject_reason_code": "E3",
+    "reject_reason_description": "Correction required - relational fields in error.",
+    "follow-up_action_code": "C",
+    "follow-up_action_description": "Please Correct and Resubmit",
```
Comment (ankitdimania-eligible) [inline src/test/resources/com/eligible/model/api_error_response.json]: check the sample here.

File: src/main/java/com/eligible/json/deserializer/ErrorDeserializer.java (hunk 1 above)
Comment (ankit819) [inline src/main/java/com/eligible/json/deserializer/ErrorDeserializer.java]: Cool.

**Comment by ankit819** [comment]
Looks good.

**Pull request merged**